### PR TITLE
[WIP] Use github.com/fsnotify/fsnotify instead of k8s.io/utils/inotify

### PR DESCRIPTION
**Pull request by wzshiming**
> The k8s.io/utils/inotify is a fork of golang.org/x/exp/inotify before it was deleted.
> Please use gopkg.in/fsnotify.v0 instead.
> For updates, see: https://fsnotify.org/

https://github.com/kubernetes/utils/tree/master/inotify

This is the last use of `k8s.io/utils/inotify` in the code base.  Maybe it's better to change to `github.com/fsnotify/fsnotify`.

#### What type of PR is this?
/kind feature
/sig node

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Use github.com/fsnotify/fsnotify instead of k8s.io/utils/inotify
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by adisky — COMMENTED** [review]
@wzshiming IMO it is not a user facing change, we are just changing a internal library usage

/triage accepted 
/priority important-longterm

File: cmd/kubelet/app/server_linux.go [hunk 1]
```diff
@@ -17,30 +17,38 @@ limitations under the License.
...
-		select {
-		case ev := <-watcher.Event:
-			klog.InfoS("inotify event", "event", ev)
-		case err = <-watcher.Error:
-			klog.ErrorS(err, "inotify watcher error")
+		defer func() {
+			close(done)
+			watcher.Close()
+		}()
+		for {
```
Comment (matthyx) [inline cmd/kubelet/app/server_linux.go:40]: @wzshiming I don't understand why we didn't have the `for` loop before... any idea?
/assign

Comment (wzshiming) [reply cmd/kubelet/app/server_linux.go:40]: The original `k8s.io/utils/inotify` can choose the registered event when `AddWatch`, but `github.com/fsnotify/fsnotify` can't, so filter after getting.

I missed `inotify.InOpen`, this event there is no watched in `github.com/fsnotify/fsnotify`
No wonder there is only one left `k8s.io/utils/inotify`

The logic here is mainly to exit Kubelet if someone opens or deletes the lockfile
https://github.com/kubernetes/kubernetes/blob/254c77bb8905b80981fa6e4dc5c98c7616f567c0/cmd/kubelet/app/server.go#L531-L543

Could I modify the logic to achieve the same effect?

**Comment by aojea** [comment]
I vaguely remember, from a PR that addressed a similar change in kube-proxy, that @liggitt @thockin mentioned there are some historical context on the inotify behaviour ... :thinking:

**Comment by wzshiming** [comment]
> I vaguely remember, from a PR that addressed a similar change in kube-proxy, that @liggitt @thockin mentioned there are some historical context on the inotify behaviour ... 🤔

The `inotify.InOpen` event there is not watched in `github.com/fsnotify/fsnotify` 😂

**Pull request closed without merging**